internal/api/registry/manifests_test.go

```diff
@@ -600,14 +600,14 @@ func TestManifestRequiredLabels(t *testing.T) {
 			ExpectHeader: test.VersionHeader,
 			ExpectBody: test.ErrorCodeWithMessage{
 				Code:    keppel.ErrManifestInvalid,
-				Message: "missing required labels: somethingelse, andalsothis",
+				Message: "rule \"has(labels.randomlabelthatdoesnotexist)\" is not satisfied",
```

Comment:
Can we somehow extract why the rule did not apply? Especially when combining boolean conditions it could be a bit confusing why exactly things failed.

Reply:
I think it's fine to just say "this rule was not satisified", and then the caller will have to figure out which part they did wrong. This is kind of unavoidable when going with generic expressions. Even the stuff that Valentin describes, while potentially useful for the classic `has(labels.foo) && has(labels.bar)` usecase, will fall apart with more sophisticated expressions. So we would have to do a lot of work to give out a diagnostic that only helps with reasonably simple cases anyway, which I don't find to be a useful tradeoff.